components/camel-sip/src/main/java/org/apache/camel/component/sip/SipConfiguration.java

```diff
@@ -53,124 +53,348 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * Stores information used for sending and receiving SIP messages.
+ */
 @UriParams
-public class SipConfiguration {    
+public class SipConfiguration {
+
+    /**
+     * The logger for this class.
+     */
     private static final Logger LOG = LoggerFactory.getLogger(SipConfiguration.class);
+
+    /**
+     * The SIP library implementation used by the SipFactory for this camel component.
+     */
     private static final String IMPLEMENTATION = "gov.nist";
 
+    /**
+     * The SipComponent which uses this SipConfiguration to create a SipEndpoint. Primarily used to
+     * help with resolving the parameter list given when creating an endpoint.
+     */
     private SipComponent component;
 
+    /**
+     * The specified SIP Protocol that should be used. Has to be equal to either "sip" or "sips" to be valid.
+     */
     private String protocol;
+
+    /**
+     * A map of parameters given in the sip URI when the SipEndpoint is created.
+     */
     private Map<String, Object> parameters;
 
+    /**
+     * The SIP URI the SipEndpoint needs to connect to.
+     * This object represents the SIP URI string given when the SipEndpoint is created.
+     */
     @UriPath @Metadata(required = "true")
     private URI uri;
-    @UriParam(label = "advanced")
-    private AddressFactory addressFactory;
```

Comment:
@nikvaessen is there a reason you changed the order of these variables?  Could you put these variables back in the same order so the diff is easier to review?  Thanks for inserting the comments about each of them.